AutoRest/AutoRest.Core/ClientModel/Parameter.cs

```diff
@@ -28,6 +28,20 @@ public Parameter()
         public string Name { get; set; }
 
         /// <summary>
+        /// Gets or sets the parameter client (explicitly defined code generation) name.
+        /// </summary>
+        public string ClientName {
```

Comment:
While conceptually this change is OK, it's not in line with our design philosophy (which is not documented :)). We try to not change Core client model for these extensions as much as possible. Instead we implement extensions inside Extensions project. In this case I would have a method inside extensions that would change the parameter.Name and property.Name if x-ms-client-name was specified.